static/app/components/tooltip.tsx

```diff
@@ -116,215 +111,262 @@ function computeOriginFromArrow(
   }
 }
 
-class Tooltip extends React.Component<Props, State> {
-  static defaultProps: DefaultProps = {
-    position: 'top',
-    containerDisplayMode: 'inline-block',
-  };
-
-  state: State = {
-    isOpen: false,
-    usesGlobalPortal: true,
-  };
-
-  async componentDidMount() {
-    if (IS_ACCEPTANCE_TEST) {
-      const TooltipStore = (await import('sentry/stores/tooltipStore')).default;
-      TooltipStore.addTooltip(this);
+function getTooltipPortal(useGlobalPortal: boolean): HTMLElement {
+  // Use a single global portal element for tooltips
+  if (useGlobalPortal) {
+    let portal = document.getElementById('tooltip-portal');
+    if (!portal) {
+      portal = document.createElement('div');
+      portal.setAttribute('id', 'tooltip-portal');
+      document.body.appendChild(portal);
     }
+    return portal;
   }
 
-  async componentWillUnmount() {
-    const {usesGlobalPortal} = this.state;
+  // Each tooltip gets it's own div element
+  const portal = document.createElement('div');
+  document.body.appendChild(portal);
+  return portal;
+}
 
-    if (IS_ACCEPTANCE_TEST) {
-      const TooltipStore = (await import('sentry/stores/tooltipStore')).default;
-      TooltipStore.removeTooltip(this);
-    }
-    if (!usesGlobalPortal) {
-      document.body.removeChild(this.getPortal(usesGlobalPortal));
-    }
+/**
+ * Codesplit loading of the TooltipStore used for visual acceptance tests
+ */
+async function getAcceptanceTestTooltipStore() {
+  const {tooltipStore} = await import('sentry/stores/tooltipStore');
+  return tooltipStore;
+}
+
+type UseTooltipInAcceptanceTestsOtpions = {
+  ignore: boolean;
+  setOpen: (open: boolean) => void;
+  setUseGlobalPortal: (useGlobalPortal: boolean) => void;
+  tooltipId: string;
+};
+
+/**
+ * When running the acceptance tests, we expose functionality to open / close
+ * all of the tooltips. This hook handles registration of tooltips in the store,
+ * with special care to unregister tooltips if they become ignored.
+ */
+function useTooltipInAcceptanceTests({
+  tooltipId,
+  setOpen,
+  setUseGlobalPortal,
+  ignore,
+}: UseTooltipInAcceptanceTestsOtpions) {
+  if (!IS_ACCEPTANCE_TEST) {
```

Comment:
I guess this works, but we should try and avoid having test code end up in our production bundle. Why do acceptance tests require a way to force open/close the tooltips btw, maybe we can add that to the comment?

Reply:
The store explains what it does, though yea maybe here would be useful too

Reply:
We have quite a bit of custom copy and other things within tooltips sometimes, previously they were for the most part largely untested (we were seeing some flake around them as well). Essentially for each screenshot in acceptance test, we take the screenshot, turn on all the tooltips on the page, take another screenshot. It helps indirectly cover all tooltip cases. In theory we could manually add all the jest tests and that would be sufficient... but until we do we probably shouldn't remove the majority of our coverage.